### PR TITLE
Misc docstring fixes for io

### PR DESCRIPTION
**Pull request by dstansby**
### Description

I discovered https://pypi.org/project/velin/, which is a small command line utility to automatically fix small/common numpydoc issues in docstrings. I've run it on `astropy`, and it seemed to catch a few things. This PR is fixes to `io`.

- I've checked all these by eye to make sure they look sensible
- I figured it would be best to open one PR/sub-module

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/io/fits/hdu/base.py [hunk 1]
```diff
@@ -275,24 +275,21 @@ def fromstring(cls, data, checksum=False, ignore_missing_end=False,
         Parameters
         ----------
         data : str, bytearray, memoryview, ndarray
-           A byte string containing the HDU's header and data.
-
```
Comment (pllim) [inline astropy/io/fits/hdu/base.py:279]: Why remove blank line?

**Review by pllim — APPROVED** [review]
Seems uncontroversial enough. Just a question about blank line removal.

**Comment by pllim** [comment]
I think RTD is queued. Maybe it will catch up in a few hours... 👀

File: astropy/io/fits/hdu/base.py (hunk 1 above)
Comment (dstansby) [reply astropy/io/fits/hdu/base.py:279]: I think this snuck in because there was also and indentation level fix - I'll revert the whitespace changes here.

**Comment by saimn** [comment]
This looks ready to merge ? (I ran CI again, all green except a matplotlib job)

**Comment by pllim** [comment]
Ugh, not mpl again. It is unrelated. Let's merge. Thanks, everyone!

**Comment by pllim** [comment]
Meh, maybe not worth the trouble backporting then.

**Pull request merged**